x/superfluid/keeper/stake_test.go

```diff
@@ -1028,6 +1028,154 @@ func (suite *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {1, 1, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000)), sdk.NewCoin("gamm/pool/1", sdk.NewInt(400000))},
+			[]uint64{1, 2},
+			[]bool{false, false},
+			[]sdk.Dec{sdk.NewDec(4000000), sdk.NewDec(6000000)},
+		},
+		{
+			"add unbonded validator",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded, stakingtypes.Unbonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}, {1, 1, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000)), sdk.NewCoin("gamm/pool/1", sdk.NewInt(400000))},
+			[]uint64{1, 2},
+			[]bool{false, false},
+			[]sdk.Dec{sdk.NewDec(4000000), sdk.NewDec(6000000)},
+		},
+		{
+			"undelegating not available lock id",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}},
+			[]sdk.Coin{sdk.NewCoin("gamm/pool/1", sdk.NewInt(600000))},
+			[]uint64{2},
+			[]bool{true},
+			[]sdk.Dec{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			bondDenom := suite.App.StakingKeeper.GetParams(suite.Ctx).BondDenom
+
+			// setup validators
+			valAddrs := suite.SetupValidators(tc.validatorStats)
+
+			denoms, _ := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20), sdk.NewDec(20)})
+
+			// setup superfluid delegations
+			_, intermediaryAccs, _ := suite.setupSuperfluidDelegations(valAddrs, tc.superDelegations, denoms)
+			suite.checkIntermediaryAccountDelegations(intermediaryAccs)
+
+			for index, lockId := range tc.superUnbondingLockIds {
+				// get intermediary account
+				accAddr := suite.App.SuperfluidKeeper.GetLockIdIntermediaryAccountConnection(suite.Ctx, lockId)
+				intermediaryAcc := suite.App.SuperfluidKeeper.GetIntermediaryAccount(suite.Ctx, accAddr)
+				valAddr := intermediaryAcc.ValAddr
+
+				lock, err := suite.App.LockupKeeper.GetLockByID(suite.Ctx, lockId)
+				if err != nil {
+					lock = &lockuptypes.PeriodLock{}
+				}
+
+				// get pre-superfluid delgations osmo supply and supplyWithOffset
+				presupply := suite.App.BankKeeper.GetSupply(suite.Ctx, bondDenom)
+				presupplyWithOffset := suite.App.BankKeeper.GetSupplyWithOffset(suite.Ctx, bondDenom)
+
+				// superfluid undelegate
+				intermediaryAcc, _, err = suite.App.SuperfluidKeeper.PartialSuperfluidUndelegate(suite.Ctx, lock.Owner, lockId, tc.undelegateAmounts[index])
+				if tc.expSuperUnbondingErr[index] {
+					suite.Require().Error(err)
+					continue
+				}
+				suite.Require().NoError(err)
+
+				// ensure post-superfluid delegations osmo supplywithoffset is the same while supply is not
+				postsupply := suite.App.BankKeeper.GetSupply(suite.Ctx, bondDenom)
+				postsupplyWithOffset := suite.App.BankKeeper.GetSupplyWithOffset(suite.Ctx, bondDenom)
+				suite.Require().False(postsupply.IsEqual(presupply), "presupply: %s   postsupply: %s", presupply, postsupply)
+				suite.Require().True(postsupplyWithOffset.IsEqual(presupplyWithOffset))
```

Comment:
Can we do a check with the supply amount using the amount we unlocked instead of simply comparing if equal or not?

Reply:
Done here https://github.com/osmosis-labs/osmosis/pull/5162/commits/0d1e2c59dc33aa6b49fb8f20d699fe28ce8d8148

Reply:
Checked the exact amounts here https://github.com/osmosis-labs/osmosis/pull/5162/commits/c20ddedc31c5d790d0baee6d6df3fc8314f163a3